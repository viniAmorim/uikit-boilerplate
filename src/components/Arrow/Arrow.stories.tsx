import React from 'react';

import { Story, Meta } from '@storybook/react';

import { Arrow, ArrowProps } from './Arrow';

export default {
  title: 'Atoms/Arrow',
  component: Arrow,
} as Meta;

const Template: Story<ArrowProps> = (args) => <Arrow {...args} />;

export const Default = Template.bind({});
Default.args = {
  value: 53,
};
