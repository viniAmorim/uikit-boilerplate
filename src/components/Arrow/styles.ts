import styled from 'styled-components';

export const ArrowContainer = styled.div`
  width: fit-content;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

export const ArrowText = styled.span`
  font-family: 'Montserrat';
  font-weight: 700;
  font-size: 14px;
  line-height: 22px;
  color: ${({ theme }) => theme.colors.white100};
`;

export const Block = styled.div`
  width: 42px;
  height: 28px;
  padding: 6px;
  border-radius: 3px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0px -10px;

  background: ${({ theme }) => theme.colors.purple700};
`;

export const Arrow = styled.div`
  border-top: 30px solid transparent;
  border-bottom: 30px solid transparent;

  border-left: 30px solid ${({ theme }) => theme.colors.purple700};
  border-top-left-radius: 3px;
  border-bottom-left-radius: 3px;
`;
