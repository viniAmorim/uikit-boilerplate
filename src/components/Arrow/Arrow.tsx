import React, { forwardRef } from 'react';

import * as Styled from './styles';

export interface ArrowProps extends React.HTMLAttributes<HTMLDivElement> {
  value: number;
}

export const Arrow = forwardRef<HTMLDivElement, ArrowProps>(function Arrow(
  { value, ...rest },
  ref,
) {
  return (
    <Styled.ArrowContainer ref={ref} {...rest}>
      <Styled.Block>
        <Styled.ArrowText>{value}%</Styled.ArrowText>
      </Styled.Block>
      <Styled.Arrow />
    </Styled.ArrowContainer>
  );
});

export default Arrow;
