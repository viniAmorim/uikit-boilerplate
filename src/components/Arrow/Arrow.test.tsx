import React from 'react';

import { render } from '~/test/utils';

import { Arrow } from './Arrow';

describe('Arrow', () => {
  it('should render correctly', () => {
    const { container, getByText } = render(<Arrow value={32} />);

    const arrowElement = container.firstChild as HTMLElement;

    expect(getByText('32%')).toBeInTheDocument();
    expect(arrowElement).toBeInTheDocument();
    expect(arrowElement).toMatchSnapshot();
  });
});
