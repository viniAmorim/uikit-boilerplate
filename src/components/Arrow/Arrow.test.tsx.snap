// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow should render correctly 1`] = `
<div
  class="sc-beqWaB kQzpQb"
>
  <div
    class="sc-dmqHEX gWVQob"
  >
    <span
      class="sc-gueYoa eVPTPq"
    >
      32
      %
    </span>
  </div>
  <div
    class="sc-hLseeU ldQUJZ"
  />
</div>
`;
